test/e2e/e2e_test.go

```diff
@@ -131,6 +132,70 @@ var _ = ginkgo.Describe("JobSet", func() {
 		})
 	})
 
+	// This test is added to test the JobSet transitions as Kueue would when:
+	// doing: resume in RF1 -> suspend -> resume in RF2.
+	// In particular, Kueue updates the PodTemplate on suspending and resuming
+	// the JobSet.
+	ginkgo.When("JobSet is suspended and resumed", func() {
+
+		ginkgo.It("should allow to resume JobSet after updating PodTemplate", func() {
+			ctx := context.Background()
+			js := sleepTestJobSet(ns, 1).Obj()
+			jsKey := types.NamespacedName{Name: js.Name, Namespace: js.Namespace}
+
+			ginkgo.By("Create a suspended JobSet", func() {
+				js.Spec.Suspend = ptr.To(true)
+				js.Spec.TTLSecondsAfterFinished = ptr.To[int32](5)
```

Comment:
why do we need to set TTL for this test?

Reply:
Not needed really, used it just to make sure the Job is deleted at some point, but we don't need to. Reverted.